recipes-bsp/u-boot/files/0033-iot2050-Enable-eMMC-RPMB-support-and-OPTee-driver.patch

```diff
@@ -0,0 +1,47 @@
+From 963a046af89befe18cce1a208e5cba10d36216ee Mon Sep 17 00:00:00 2001
+From: Baocheng Su <baocheng.su@siemens.com>
+Date: Tue, 27 Sep 2022 10:21:48 +0800
+Subject: [PATCH 33/33] iot2050: Enable eMMC RPMB support and OPTee driver
+
+By enabling the eMMC RPMB support and the OPTee driver, UEFI secure boot
+can take advantage of the RPMB based secure storage provided by OPTee.
+
+This also prepares the RPMB otp key programming via U-Boot by enabling
+the `CONFIG_CMD_OPTEE_RPMB`.
+
+For security concerns, by default disable the RPMB operation in the
+`mmc` command.
```

Comment:
What does this mean? Note that there is no console in secure boot mode. And if things should be off in that mode, there is an extra config snippet to flip controls.

Reply:
Also: The U-Boot RPMB commands are purely containing operations for initial deployment? If not, that would be another reason to keep them enabled in non-secure builds.

Reply:
There is nothing to do with secure boot.

the u-boot `mmc` command also exposes some rpmb sub-commands for read/write/key the RPMB by enabling the `CONFIG_CMD_OPTEE_RPMB` which is implied by `CONFIG_SUPPORT_EMMC_RPMB` :

```
config SUPPORT_EMMC_RPMB
	bool "Support eMMC replay protected memory block (RPMB)"
	imply CMD_MMC_RPMB
```

Esp. for the `mmc rpmb key` sub command, it can program anything as the otp key into the RPMB, if such command is exposed to our pre-installed u-boot, then in some cases (such as on the way shipping), somebody could easily take the advantage of this command to flash anything into the RPMB otp, to induce a RPMB subsystem Denial-of-Service.

Reply:
A better way is to use a separate CONFIG_ for the `mmc rpmb key` sub-command.

Reply:
> Also: The U-Boot RPMB commands are purely containing operations for initial deployment? If not, that would be another reason to keep them enabled in non-secure builds.

If RPMB key is not programmed, then this `mmc rpmb` command could not do anything but programming the key. However,  this command does not generate key but read the key from somewhere in the memory, this is not what we suggest to do.
 
U-Boot provides two command to manipulate the RPMB, what we should use is the other `optee-rpmb` command which generates the key inside optee.

Reply:
So even for non-secure build, to reduce the chance of misoperation, the rpmb sub command provided by `mmc` should be disabled by default.

Reply:
For the read/write operations, since this command use memory key instead of the key generated in OPTee, it is totally useless on our board.

Reply:
Accepted as safety measure, rejected as pseudo security measure (every attacker can install anything as long as the device is not in secure mode).

Reply:
what about change `For security concerns, by default disable the RPMB operation in the
mmc command.` 

to `Disable the mmc rpmb sub command to aviod mis-programming of the RPMB key `

Reply:
Or `Disable the mmc rpmb sub command to reduce the chance of mis-programming of the RPMB key`

Reply:
Ack.